### PR TITLE
Adding support for migrating paragraph types + Migration lookup.

### PR DESCRIPTION
**Pull request by joshirohit100**
cc @amitgoyal @pranitjha for review.

### CODE REVIEW

File: src/Plugin/DestinationField/EntityReferenceRevision.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Drupal\migration_glue\Plugin\DestinationField;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\migration_mapper\DestinationFieldBase;
+
+/**
+ * Provides an 'image' destination field type.
```
Comment (amitgoyal) [inline src/Plugin/DestinationField/EntityReferenceRevision.php]: image -> paragraphs or similar entities

File: src/Plugin/DestinationField/EntityReferenceRevision.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  public function getExport(string $field_name, array $field_data, FormStateInterface $form_state) {
+    $field_name = $this->overrideName($field_name);
+    return [
+      $field_name => trim($field_data['map_to']),
+    ];
+  }
+
+  /**
+   * Gives overridden field name.
+   *
```
Comment (amitgoyal) [inline src/Plugin/DestinationField/EntityReferenceRevision.php]: Some example code

File: src/Plugin/DestinationField/TextWithSummaryExtended.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  public function getExport(string $field_name, array $field_data, FormStateInterface $form_state) {
+    $field_name = $this->overrideName($field_name);
+    return [
+      $field_name => trim($field_data['map_to']),
+    ];
+  }
+
+  /**
+   * Gives overridden field name.
+   *
```
Comment (amitgoyal) [inline src/Plugin/DestinationField/TextWithSummaryExtended.php]: Add sample example code

File: src/Plugin/FieldProcessor/MigrationLookup.php [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      '#title' => $this->t('No stub'),
+      '#options' => [
+        1 => $this->t('Yes'),
+        0 => $this->t('No'),
+      ],
+    ];
+
+    $form['use_extract'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Use extract'),
```
Comment (amitgoyal) [inline src/Plugin/FieldProcessor/MigrationLookup.php]: may be help text.

File: src/Plugin/FieldProcessor/MigrationLookup.php [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    $form['migration'] = [
+      '#type' => 'select',
+      '#description' => $this->t('Please select migration.'),
+      '#title' => $this->t('Migration'),
+      '#options' => $this->getMigrationList(),
+    ];
+
+    $form['no_stub'] = [
+      '#type' => 'select',
+      '#description' => $this->t('No stub.'),
```
Comment (amitgoyal) [inline src/Plugin/FieldProcessor/MigrationLookup.php]: may be add help text

File: src/Plugin/FieldProcessor/MigrationLookup.php [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+      '#title' => $this->t('Use extract'),
+      '#options' => [
+        1 => $this->t('Yes'),
+        0 => $this->t('No'),
+      ],
+    ];
+
+    $form['index'] = [
+      '#type' => 'number',
+      '#description' => $this->t('Please provide the index position like 0,1 etc.'),
```
Comment (amitgoyal) [inline src/Plugin/FieldProcessor/MigrationLookup.php]: may be add help text indicating paragraphs target_id and target_revision_id

File: src/Plugin/FieldProcessor/MigrationLookup.php [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+  /**
+   * Get list of migration.
+   *
+   * @return array
+   */
+  protected function getMigrationList() {
+    // @TODO: Load migration instead of hardcoded.
+    return [
+      'example_id' => 'example_id',
```
Comment (amitgoyal) [inline src/Plugin/FieldProcessor/MigrationLookup.php]: todo: make it dynamic

**Comment by amitgoyal** [comment]
Mostly looks good!

File: src/Plugin/DestinationField/EntityReferenceRevision.php (hunk 1 above)
Comment (joshirohit100) [reply src/Plugin/DestinationField/EntityReferenceRevision.php]: PR updated! thanks!

File: src/Plugin/DestinationField/EntityReferenceRevision.php (hunk 2 above)
Comment (joshirohit100) [reply src/Plugin/DestinationField/EntityReferenceRevision.php]: PR updated! thanks!

File: src/Plugin/DestinationField/TextWithSummaryExtended.php (hunk 3 above)
Comment (joshirohit100) [reply src/Plugin/DestinationField/TextWithSummaryExtended.php]: PR updated! thanks!

File: src/Plugin/FieldProcessor/MigrationLookup.php (hunk 4 above)
Comment (joshirohit100) [reply src/Plugin/FieldProcessor/MigrationLookup.php]: PR updated! thanks!

File: src/Plugin/FieldProcessor/MigrationLookup.php (hunk 5 above)
Comment (joshirohit100) [reply src/Plugin/FieldProcessor/MigrationLookup.php]: PR updated! thanks!

File: src/Plugin/FieldProcessor/MigrationLookup.php (hunk 6 above)
Comment (joshirohit100) [reply src/Plugin/FieldProcessor/MigrationLookup.php]: PR updated! thanks!

File: src/Plugin/FieldProcessor/MigrationLookup.php (hunk 7 above)
Comment (joshirohit100) [reply src/Plugin/FieldProcessor/MigrationLookup.php]: PR updated! Thanks!

**Pull request merged**